### ISSUE TITLE
literal number values in where queries

### ISSUE THREAD

**Issue by EricSch** [Breeze/breeze.js#93]
It seems Breeze converts a literal number always to DataType.Double. See function DataType.fromValue.
This gives problem (OData exception) when the first parameter, usually a data field, has another number type, like decimal.

Why not reflecting the number type of the first parameter and convert the literal number to that type?

Thanks
Eric

**Comment by EricSch** [comment]
It is used with the Kendo Grid and filtering enabled.
So a filter is send to breeze like that
`Object {field: "surfaceBuilding", operator: "gt", value: 240}`

Without forcing the value, it would be of `DataType` Float.

>EntityQuery {resourceName: "ParzelleSearch", fromEntityType: null, wherePredicate:
AndOrPredicate, orderByClause: ctor, selectClause: null…}
dataService: undefined
entityManager: null
expandClause: null
fromEntityType: null
inlineCountEnabled: true
noTrackingEnabled: false
orderByClause: ctor
parameters: Object
queryOptions: undefined
resourceName: "ParzelleSearch"
resultEntityType: undefined
selectClause: null
skipCount: null
takeCount: 10
usesNameOnServer: undefined
wherePredicate: AndOrPredicate
__proto__: EntityQuery

Shouldn't breeze able to get the metadata of the query `ParzelleSearch` and get the type of the field `surfaceBuilding`?

**Comment by wardbell** [comment]
Breeze would only know how to get the type info if "PazelleSearch" is a resource name it can associate with an `EntityType`.

My guess is that "PazelleSearch" is not a known resource mame. Fortunately, you can register that name with an `EntityType`. See the API doc for the `Metadata` class.

**Comment by EricSch** [comment]
Would that solve the problem?

"ParzelleSearch" returns the entity "ParzelleSearchResult", and that entitiy is already known,

**Comment by wardbell** [comment]
Yes I think so. Just register "ParzelleSearch" as one of the resource names associated with the "ParzelleSearchResult" entity type and I think all will be well. Let us know. If so, we can close this issue.

**Comment by cdt3pm** [comment]
Hi, I have encountered a similar problem because my `EntityType` was not mapped. I agree that the best way to avoid this problem is to map the entity type, but I still think the `DataType.fromValue` should use the built-in function [Number.isInteger](https://www.w3schools.com/jsref/jsref_isinteger.asp), especially because `fromValue` is public. Here is the code from 1.7.1:
```
DataType.fromValue = function (val) {
    if (__isDate(val)) return DataType.DateTime;
    switch (typeof val) {
      case "string":
        if (__isGuid(val)) return DataType.Guid;
        // the >3 below is a hack to insure that if we are inferring datatypes that
        // very short strings that are valid but unlikely ISO encoded Time's are treated as strings instead.
        else if (__isDuration(val) && val.length > 3) return DataType.Time;
        else if (__isDateString(val)) return DataType.DateTime;
        return DataType.String;
      case "boolean":
        return DataType.Boolean;
      case "number":
        return DataType.Double;
    }
    return DataType.Undefined;
  };
`
